flink-connectors/flink-connector-mongodb/src/main/java/org/apache/flink/mongodb/table/MongodbDynamicTableSourceSinkFactory.java

```diff
@@ -0,0 +1,133 @@
+package org.apache.flink.mongodb.table;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.mongodb.table.sink.MongodbDynamicTableSink;
+import org.apache.flink.mongodb.table.sink.MongodbSinkConf;
+import org.apache.flink.mongodb.table.source.MongodbDynamicTableSource;
+import org.apache.flink.mongodb.table.util.ContextUtil;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.connector.format.DecodingFormat;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DeserializationFormatFactory;
+import org.apache.flink.table.factories.DynamicTableSinkFactory;
+import org.apache.flink.table.factories.DynamicTableSourceFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.utils.TableSchemaUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class MongodbDynamicTableSourceSinkFactory implements DynamicTableSinkFactory, DynamicTableSourceFactory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MongodbDynamicTableSourceSinkFactory.class);
+    @VisibleForTesting
+    public static final String IDENTIFIER = "mongodb";
+    public static final ConfigOption<String> DATABASE = ConfigOptions.key("database".toLowerCase())
```

Comment:
Please move the config options and the factory identifier to a `*ConnectorOptions` class (refer to e.g. `KafkaConnectorOptions`) and mark that class `@PublicEvolving`. Also, that class and the entire connector should live in a `o.a.f.connector.mongodb` package for consistency, and the class with the options in `o.a.f.connector.mongodb.table`.

Reply:
> 不支持upsert么

@dailai The connector supports two modes, "insert" and "merge into". Merge into is similar to what you call update. It is used when the primary key is unique, there is update for the primary key, and there is no insert for the primary key